src/handler.ts

```diff
@@ -73,14 +80,58 @@ const jumpTable: JumpTable = {
     },
   },
   schema: {
-    Create: async (resourceId: string) => {
-      return format("create schema if not exists %I", resourceId)
+    Create: async (resourceId: string, props: SchemaProps) => {
+      const sql: string[] = [format("create schema if not exists %I", resourceId)]
+      if (props.RoleName) {
+        sql.push(format("GRANT USAGE ON SCHEMA %I TO %I", resourceId, props.RoleName))
+        sql.push(format("GRANT CREATE ON SCHEMA %I TO %I", resourceId, props.RoleName))
+        sql.push(
+          format("GRANT ALL ON ALL TABLES IN SCHEMA %I TO %I", resourceId, props.RoleName)
+        )
+      }
+      return sql
     },
-    Update: async (resourceId: string, oldResourceId: string) => {
-      return format("alter schema %I rename to %I", oldResourceId, resourceId)
+    Update: async (resourceId: string, oldResourceId: string, props: SchemaProps) => {
+      const sql: string[] = []
+      // TODO: revoke old role-name if props.RoleName was removed or changed
+      if (props.RoleName) {
+        sql.push(
+          format(
+            "REVOKE ALL ON ALL TABLES IN SCHEMA %I FROM %I",
+            oldResourceId,
+            props.RoleName
+          )
+        )
+        sql.push(
+          format("REVOKE CREATE ON SCHEMA %I FROM %FROM", oldResourceId, props.RoleName)
+        )
+        sql.push(format("REVOKE ALL ON SCHEMA %I FROM %I", oldResourceId, props.RoleName))
+      }
+      sql.push(format("alter schema %I rename to %I", oldResourceId, resourceId))
+      if (props.RoleName) {
+        sql.push(format("GRANT USAGE ON SCHEMA %I TO %I", resourceId, props.RoleName))
+        sql.push(format("GRANT CREATE ON SCHEMA %I TO %I", resourceId, props.RoleName))
+        sql.push(
+          format("GRANT ALL ON ALL TABLES IN SCHEMA %I TO %I", resourceId, props.RoleName)
+        )
+      }
+      return sql
     },
-    Delete: (resourceId: string) => {
-      return format("drop schema if exists %I cascade", resourceId)
+    Delete: (resourceId: string, props: SchemaProps) => {
+      const sql: string[] = []
+      if (props.RoleName) {
+        sql.push(
```

Comment:
You're repeating yourself a lot, so please put the grant + revoke schema into their own function, so you can call that instead of copying the same code.

Reply:
You're doing the same in may other places:

https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L100
https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L145

and

https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L187
https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L211

and

https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L211
https://github.com/berenddeboer/cdk-rds-sql/blob/main/src/handler.ts#L140

However, I moved the statements into functions.